agent/sidecar_service.go

```diff
@@ -69,6 +69,14 @@ func (a *Agent) sidecarServiceFromNodeService(ns *structs.NodeService, token str
 		sidecar.Tags = append(sidecar.Tags, ns.Tags...)
 	}
 
+	// Copy weights from the original service
+	if ns.Weights != nil {
```

Comment:
Similar to some of the conditions above I guess this should also check that `sidecar.Weights == nil` so that we don't overwrite existing weights if they are set.

Reply:
Hi @dnephin . I just checked it, and sidecar.Weights are never nil. When creating the service definition from config it is set safe defaults to 1 and 1 (passing and warning) and they remain with those values with no way to modify them with the current code.

Reply:
Ah yes, good point. I'm not sure what happens if the service is created from the API.

It might make sense to move those defaults to be closer where they are used, so that we can tell during config loading if they are set or not. I have not looked to see how practical that change would be.